### PR TITLE
Task/receive notifications

### PR DESCRIPTION
**Pull request by marccolm**
#### What's this PR do?
Receives notifications on users, clients, contacts, igualas, and works. Shows toast notifications for works changes, and shows custom errors on users, igualas and contacts on delete.
#### Where should the reviewer start?
Users, clients, contacts, igualas, works components.

#### How should this be manually tested?
Displaying lists and making changes on the admin so the banner is shown.

#### Any background context you want to provide?

This template was adapted ~stolen~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

### CODE REVIEW

File: src/app/accounts/clients/clients.component.ts [hunk 1]
```diff
@@ -179,4 +192,28 @@ export class ClientsComponent implements OnInit {
...
+
+  /**
+  * Opens websocket connection to specified url and
+  * to receive notification of changes to the database.
+  * Params:
+  *   - userId: Current user id to connect to correct channel
+  *   - url: General address to connect to, to receive notifications
+  **/
+  public receiveNotifications(url: string) {
+    let ws_path = environment.WS_URL + url;
```
Comment (EduardoVaca) [inline src/app/accounts/clients/clients.component.ts]: Declare variable as `wsPath` is more Javascript appropriate

File: src/app/accounts/contacts/contacts.component.ts [hunk 2]
```diff
@@ -205,4 +223,28 @@ export class ContactsComponent implements OnInit {
...
+
+  /**
+  * Opens websocket connection to specified url and
+  * to receive notification of changes to the database.
+  * Params:
+  *   - userId: Current user id to connect to correct channel
+  *   - url: General address to connect to, to receive notifications
+  **/
+  public receiveNotifications(url: string) {
+    let ws_path = environment.WS_URL + url;
```
Comment (EduardoVaca) [inline src/app/accounts/contacts/contacts.component.ts]: Declare variable as wsPath is more Javascript appropriate

File: src/app/admin/users/users.component.ts [hunk 3]
```diff
@@ -192,4 +211,29 @@ export class UsersComponent implements OnInit {
...
+
+  /**
+  * Opens websocket connection to specified url
+  * to receive notification of changes to the database.
+  * Params:
+  *   - userId: Current user id to connect to correct channel
+  *   - url: General address to connect to, to receive notifications
+  **/
+  public receiveNotifications(url: string) {
+    let ws_path = environment.WS_URL + url;
```
Comment (EduardoVaca) [inline src/app/admin/users/users.component.ts]: Declare variable as wsPath is more Javascript appropriate

Comment (fernandolobato) [inline src/app/admin/users/users.component.ts]: ya duermanse

File: src/app/general-user/works/works.component.ts [hunk 4]
```diff
@@ -89,6 +96,15 @@ export class WorksComponent implements OnInit {
     this.loadWorkTypesForGraduation(environment.ART_TYPES_URL);
     this.loadStatusList(environment.STATUS_URL);
     this.loadUserExecutivesList(environment.USERS_URL);
+    this.notificationBannerIsActive = false;
+    this.userId = this.getCurrentUserId();
```
Comment (EduardoVaca) [inline src/app/general-user/works/works.component.ts]: Remove this variable

File: src/app/general-user/works/works.component.ts [hunk 5]
```diff
@@ -89,6 +96,15 @@ export class WorksComponent implements OnInit {
...
+    this.receiveNotifications(environment.WORK_LIST_NOTIFICATIONS_URL, this.userId);
```
Comment (EduardoVaca) [inline src/app/general-user/works/works.component.ts]: replace for:
`this.receiveNotifications(environment.WORK_LIST_NOTIFICATIONS_URL, this.httpService.getCurrentUser().id);`

File: src/app/general-user/works/works.component.ts [hunk 6]
```diff
@@ -333,8 +349,87 @@ export class WorksComponent implements OnInit {
...
+    this.loadWorkTypesList(environment.WORK_TYPES_URL);
+    this.loadWorkTypesForGraduation(environment.ART_TYPES_URL);
+    this.loadStatusList(environment.STATUS_URL);
+    this.loadUserExecutivesList(environment.USERS_URL);
+  }
+
+  /**
+  * Returns current user's id.
+  **/
+  public getCurrentUserId() {
```
Comment (EduardoVaca) [inline src/app/general-user/works/works.component.ts]: Remove this method

File: src/app/general-user/works/works.component.ts [hunk 7]
```diff
@@ -333,8 +347,75 @@ export class WorksComponent implements OnInit {
...
+    this.loadContactsList(environment.CONTACTS_URL);
+    this.loadIgualasList(environment.IGUALAS_URL);
+    this.loadWorkTypesList(environment.WORK_TYPES_URL);
+    this.loadWorkTypesForGraduation(environment.ART_TYPES_URL);
+    this.loadStatusList(environment.STATUS_URL);
+    this.loadUserExecutivesList(environment.USERS_URL);
+  }
+
+  /**
+  * Creates a toast depending on a Response object.
```
Comment (EduardoVaca) [inline src/app/general-user/works/works.component.ts]: Update comments...

File: src/app/general-user/works/works.component.ts [hunk 8]
```diff
@@ -333,8 +347,75 @@ export class WorksComponent implements OnInit {
...
+        type: type,
+        title: title,
+        body: message,
+        showCloseButton: false
+    };
+    return toast;
+  }
+
+  /**
+  * Shows a toast depending on a Response object.
```
Comment (EduardoVaca) [inline src/app/general-user/works/works.component.ts]: Update comments

**Pull request merged**